staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/validator.go

```diff
@@ -99,11 +111,59 @@ func (v *validator) Validate(versionedAttr *generic.VersionedAttributes, version
 			} else {
 				decision.Message = fmt.Sprintf("failed expression: %v", strings.TrimSpace(validation.Expression))
 			}
-
 		} else {
 			decision.Action = ActionAdmit
 			decision.Evaluation = EvalAdmit
 		}
 	}
-	return decisions
+
+	auditAnnotationEvalResults, err := v.auditAnnotationFilter.ForInput(versionedAttr, versionedParams, cel.CreateAdmissionRequest(versionedAttr.Attributes))
+	if err != nil {
+		return ValidateResult{
+			Decisions: []PolicyDecision{
+				{
+					Action:     policyDecisionActionForError(f),
+					Evaluation: EvalError,
+					Message:    err.Error(),
+				},
+			},
+		}
+	}
+
+	auditAnnotationResults := make([]PolicyAuditAnnotation, len(auditAnnotationEvalResults))
+	for i, evalResult := range auditAnnotationEvalResults {
+		var auditAnnotationResult = &auditAnnotationResults[i]
+		// TODO: move this to generics
+		validation, ok := evalResult.ExpressionAccessor.(*AuditAnnotationCondition)
```

Comment:
Is that possible that `evalResult.ExpressionAccessor` is nil and `evalResult.Error` saved the err? 
Looks at the code path checks evalResult.ExpressionAccessor.(*AuditAnnotationCondition) before `evalResult.Error` and the error might be ignored?

Reply:
This is the same order as for `validations` above. This `!ok` check is a failure that should never be reached if the code is working as expected (and will go away when we switch to using generics here).